client/src/Movies/Movie.js

```diff
@@ -26,42 +29,30 @@ export default class Movie extends Component {
       });
   };
   // Uncomment this code when you're ready for the stretch problems
-  // componentWillReceiveProps(newProps){
-  //   if(this.props.match.params.id !== newProps.match.params.id){
-  //     this.fetchMovie(newProps.match.params.id);
-  //   }
-  // }
+   componentWillReceiveProps(newProps){
+     if(this.props.match.params.id !== newProps.match.params.id){
+      this.fetchMovie(newProps.match.params.id);
+    }
+  }
 
-  // saveMovie = () => {
-  //   const addToSavedList = this.props.addToSavedList;
-  //   addToSavedList(this.state.movie)
-  // }
+   saveMovie = () => {
+     const addToSavedList = this.props.addToSavedList;
+    addToSavedList(this.state.movie)
```

Comment:
So, you've got an error going on here saying "addToSavedList is not a function." Adding one little thing somewhere in your code (hint: not here) fixes this error and makes your saved list functionality work perfectly. Let me know when you get it fixed :)